### DISCUSSION TITLE
@MainActor on XCTestCase warns with strict concurrency checking

### DISCUSSION THREAD

**Discussion by humblehacker** [pointfreeco/swift-composable-architecture#2739] [Q&A] [answered]
I've turned on strict concurrency checking. Currently using the Xcode 15.3 beta. I have a test case that I defined following [the documentation](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/testing/#Testing-state-changes). With strict concurrency checking I get the following warning: 

> Main actor-isolated class 'ContentReducerTests' has different actor isolation from nonisolated superclass 'XCTestCase'; this is an error in Swift 6

<img width="660" alt="Screenshot 2024-01-29 at 1 15 48 PM" src="https://github.com/pointfreeco/swift-composable-architecture/assets/117582/33fb60c3-426d-444c-a330-75306565836b">

Should I ignore this?

I noticed that if I move the `@MainActor` attribute to the test functions instead of the whole class, the warning goes away. But I'm uncertain if this is correct.

**Comment by stephencelis** [comment, marked as answer]
@humblehacker You can move the `@MainActor` annotations to individual test cases instead to work around this warning, but it may be worth bringing up on the Swift forums or in a feedback!
